client/my-sites/plans/jetpack-plans/upsell/index.tsx

```diff
@@ -0,0 +1,176 @@
+import {
+	PRODUCT_JETPACK_BACKUP_T1_YEARLY,
+	PRODUCT_JETPACK_SCAN,
+	PRODUCT_JETPACK_ANTI_SPAM,
+	PLAN_JETPACK_SECURITY_T1_YEARLY,
+	getJetpackProductsDisplayNames,
+} from '@automattic/calypso-products';
+import { Button } from '@automattic/components';
+import { translate } from 'i18n-calypso';
+import page from 'page';
+import { useEffect, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import QueryIntroOffers from 'calypso/components/data/query-intro-offers';
+import QuerySiteProducts from 'calypso/components/data/query-site-products';
+import Main from 'calypso/components/main';
+import { useExperiment } from 'calypso/lib/explat';
+import PlanPrice from 'calypso/my-sites/plan-price';
+import { buildCheckoutURL } from 'calypso/my-sites/plans/jetpack-plans/get-purchase-url-callback';
+import useItemPrice from 'calypso/my-sites/plans/jetpack-plans/use-item-price';
+import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
+import { getSitePurchases, isFetchingSitePurchases } from 'calypso/state/purchases/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { PURCHASE_FLOW_UPSELLS_MATRIX } from '../constants';
+import slugToSelectorProduct from '../slug-to-selector-product';
+import type { QueryArgs } from 'calypso/my-sites/plans/jetpack-plans/types';
+
+import './style.scss';
+
+interface Props {
+	siteSlug: string;
+	productSlug: string;
+	urlQueryArgs: QueryArgs;
+}
+
+const JetpackUpsellPage: React.FC< Props > = ( { siteSlug, productSlug, urlQueryArgs } ) => {
+	const upsellSlug = PURCHASE_FLOW_UPSELLS_MATRIX[ productSlug ];
+	const productNames = getJetpackProductsDisplayNames();
+
+	const [ isLoadingUpsellPageExperiment, experimentAssignment ] = useExperiment(
+		'jetpack_upsell_page_2022_05'
+	);
+
+	const siteId = useSelector( ( state ) => getSelectedSiteId( state ) );
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const isFetchingPurchases = useSelector( isFetchingSitePurchases );
+	const purchases = useSelector( ( state ) => getSitePurchases( state, siteId ) );
+
+	const productItem = useMemo( () => slugToSelectorProduct( productSlug ), [ productSlug ] );
+	const upsellItem = useMemo(
+		() => ( upsellSlug ? slugToSelectorProduct( upsellSlug ) : null ),
+		[ upsellSlug ]
+	);
+	const productCheckoutURL = useMemo(
+		() => buildCheckoutURL( siteSlug, productSlug, urlQueryArgs ),
```

Comment:
Intuitively, it seems that the close button in the checkout page is not taking into account the `checkoutBackUrl` query parameter. We'd need to confirm that's the actual issue.

This might soon not be relevant anymore: p1HpG7-gg7-p2.

Reply:
Resolving in favor of addressing in a separate PR 👍